platform/darwin/src/NSComparisonPredicate+MGLAdditions.mm

```diff
@@ -0,0 +1,76 @@
+#import "NSComparisonPredicate+MGLAdditions.h"
+
+#import "NSPredicate+MGLAdditions.h"
+#import "NSExpression+MGLAdditions.h"
+
+class FilterEvaluator {
+public:
+
+mbgl::style::Filter operator()(NSComparisonPredicate *predicate) {
```

Comment:
This function should be inlined into `-[NSComparisonPredicate(MGLAdditions) mgl_filter]`.